kyo-prelude/shared/src/main/scala/kyo/Parse.scala

```diff
@@ -58,8 +65,8 @@ object Parse:
       * @return
       *   Parsed value if successful, drops the current parse branch if unsuccessful
       */
-    def read[A](f: Text => Maybe[(Text, A)])(using Frame): A < Parse =
-        Var.use[State] { state =>
+    def read[A, In](f: Chunk[In] => Maybe[(Chunk[In], A)])(using Frame, StateTag[In]): A < Parse[In] =
+        Var.use[State[In]] { state =>
```

Comment:
```suggestion
    def read[In](using Frame)[A](f: Chunk[In] => Maybe[(Chunk[In], A)])(using StateTag[In]): A < Parse[In] =
        Var.use[State[In]] { state =>
```

but I am not sure about it ...